### PR TITLE
Add `testing` endpoint to emily api

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Description

Closes: #375 

Adds a `/testing/wipe` endpoint to the Emily API that wipes the databases.

## Changes

Adds a `/testing` endpoint that is only created when the `testing` feature is activated which can wipe the databases.

The following are added in this PR:
- A database accessor function that wipes a database
- A database accessor that gets all entries in a table
- A `testing` endpoint route
- A `testing` endpoint route handler

## Testing Information

Tested locally by populating the database with the sequence:

```bash
# Terminal 1
make emily-integration-env-up
```

```bash
# Terminal 2
make emily-populate-database

# Get deposits and withdrawals.
curl "localhost:3000/withdrawal?status=pending&pageSize=1000" | jq
curl "localhost:3000/deposit?status=pending&pageSize=1000" | jq

curl -X POST "localhost:3000/testing/wipe"

# Get deposits and withdrawals.
curl "localhost:3000/withdrawal?status=pending&pageSize=1000" | jq
curl "localhost:3000/deposit?status=pending&pageSize=1000" | jq
```

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: emily/handler/src/api/handlers/testing.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+)]
+pub async fn wipe_databases(context: EmilyContext) -> impl warp::reply::Reply {
+    // Internal handler so `?` can be used correctly while still returning a reply.
+    async fn handler(context: EmilyContext) -> Result<impl warp::reply::Reply, Error> {
+        accessors::wipe_all_tables(&context).await?;
+        Ok(warp::reply::with_status(
+            warp::reply::json(&json!({
+                "message": "wiped database"
+            })),
+            StatusCode::NO_CONTENT,
```
Comment (djordon) [inline emily/handler/src/api/handlers/testing.rs:34]: I know this is a test endpoint so it "doesn't matter", but still I figured that I should point out that you probably shouldn't bother returning anything with a `204 No Content` response. I'm under the impression that many clients don't attempt to parse the body.

File: emily/handler/src/api/routes/mod.rs [hunk 2]
```diff
@@ -11,13 +11,30 @@ mod chainstate;
...
+        health::routes()
+            .or(chainstate::routes(context.clone()))
+            .or(deposit::routes(context.clone()))
+            .or(withdrawal::routes(context.clone()))
+            .or(testing::routes(context)),
+    )
+}
+
+#[cfg(not(feature = "testing"))]
+fn debug_routes() -> Option<impl Filter<Extract = impl warp::Reply, Error = warp::Rejection>> {
```
Comment (djordon) [inline emily/handler/src/api/routes/mod.rs:37]: Shouldn't this be `routes` with the other one called `debug_routes`?

**Review by djordon — APPROVED** [review]
Looks good. I had some minor comments. Oh and I think you need to `cargo fmt` too.

File: emily/handler/src/api/routes/mod.rs (hunk 2 above)
Comment (AshtonStephens) [reply emily/handler/src/api/routes/mod.rs:37]: fixing!

File: emily/handler/src/api/handlers/testing.rs (hunk 1 above)
Comment (AshtonStephens) [reply emily/handler/src/api/handlers/testing.rs:34]: Fair point.

File: emily/handler/src/api/routes/mod.rs (hunk 2 above)
Comment (AshtonStephens) [reply emily/handler/src/api/routes/mod.rs:37]: This was entirely wrong; both should have been called `routes` and had the same function signature.

**Pull request merged**